airflow/sensors/external_task_sensor.py

```diff
@@ -149,38 +148,22 @@ def poke(self, context, session=None):
             dttm = context['execution_date']
 
         dttm_filter = dttm if isinstance(dttm, list) else [dttm]
```

Comment:
Btw. this is something that bothers me. We "believe" in operator atomicity but this functionality allows users to provide a function that returns a list of dates. If list is provided then we query not for single DAG/task but for multiple ones.

But users will get 1/0 response - task/dag failed. there will be not information how many instances failed or for which dates. I think that the proper Airflow approach would be to have multiple sensors each checking single exec date.

@kaxil @XD-DENG WDYT?

Reply:
The use case I can think of to justify this setting-up/design is that a user is checking & depending a series of external DAG/Task.  For example, I want to depend on DAG A, and what I want to ensure is DAG A has succeeded in the past a few consecutive days (or maybe more complex case, not consecutive). Using Multiple Sensors may multiply the resource usage and may make the DAG authoring more complex in this scenario?

From another perspective, if someone prefers to have multiple sensors each checking single exec date, the current design can meet his/her requirement as well.

So personally this doesn't bother me (but as you mentioned, not giving information about how many TI failed or failed for which dates is not nice enough)

Reply:
Create a back-log ticket maybe to improve the design further or a flag to have one or the other option

Reply:
Created https://github.com/apache/airflow/issues/12577